New home page

live-wire opened the pull request:
This is in reference to issue #46 
I created a responsive layout for the home page 🧀

You can find a video of how the page looks [here](https://live-wire.github.io/evalai.webm). (size ~ 18MB) 🎥 
You can find a gif of how the page looks [here](https://live-wire.github.io/evalai.gif). (size ~ 40MB) 🖼 

Please have a look whenever you find time @deshraj @RishabhJain2018

live-wire committed:
Added the header section

live-wire committed:
Added the header section

live-wire committed:
Header minor updates

live-wire committed:
Header minor updates

live-wire committed:
Added Partner Organisations

live-wire committed:
Added Partner Organisations

live-wire committed:
Added the rules section

live-wire committed:
Added the rules section

live-wire committed:
Added Challenges section

live-wire committed:
Added Challenges section

live-wire committed:
Footer section and finishing touches

live-wire committed:
Footer section and finishing touches

live-wire committed:
Bug Fix global td css conflict

live-wire committed:
Bug Fix global td css conflict

live-wire committed:
Commented irrelevant test case to make e2e pass

live-wire committed:
Commented irrelevant test case to make e2e pass

spyshiv commented on e2e/app.e2e-spec.ts:
```diff
@@ -7,8 +7,10 @@ describe('evalai App', () => {
     page = new AppPage();
   });
 
+  /** Commenting this, since this is not relevant anymore 
```
@live-wire you may also implement the e2e test for home page.

spyshiv commented on src/app/home/home.component.html:
```diff
@@ -1,3 +1,266 @@
+<link href="https://fonts.googleapis.com/css?family=Open+Sans" rel="stylesheet">
+<!--HEADER SECTION START-->
+<div class="header_container" style="background-image: url('assets/images/Header/Header_bg.png');">
+	<div class = "row">
```
Remove the extra spaces from everywhere in the code

spyshiv commented on src/app/home/home.component.html:
```diff
@@ -1,3 +1,266 @@
+				<span class = "col-lg-1 col-md-2 col-sm-2 col-xs-12 navlinks"> News </span>
+				<span class = "col-lg-1 col-md-2 col-sm-2 col-xs-12 navlinks"> Forum </span>
+				<span class = "col-lg-1 col-md-2 col-sm-2 col-xs-12 navlinks"> Join Us </span>
+				<span class = "col-lg-1 col-md-2 col-sm-2 col-xs-12 " style = "text-align: center;"><img src="assets/images/Header/dots.png" style="height:40px;transform: rotate(90deg);text-align: center;" /></span>
```
You should not write inline css. It's not a good practice.

spyshiv commented on src/app/home/home.component.html:
```diff
@@ -1,3 +1,266 @@
+	<div class="row" style = "text-align:center">
+		<span class="section_title">Who uses EvalAI?</span>
+	</div>
+	<br><br><br>
```
Avoid using `<br>` use css property `margin, padding` etc instead.

spyshiv commented on src/app/home/home.component.scss:
```diff
@@ -0,0 +1,288 @@
+*{
```
format the whole code as per standard guidelines

spyshiv commented:
@live-wire 
## UI Review:
* There is a horizontal slider(`overflow-x`) of whole page. You should fix that.

![screenshot from 2018-03-26 04 02 14](https://user-images.githubusercontent.com/7015220/37880791-5403241e-30ab-11e8-8092-93c0fe8a5a46.png)
<br>
---

* the content here aren't center aligned
* the middle image should be same size as other images. Only on hovering the `box-shadow` should appear

![screenshot from 2018-03-26 04 05 20](https://user-images.githubusercontent.com/7015220/37880834-29bec914-30ac-11e8-946d-e8edf842ba82.png)
<br>
---

* The content alignment is not perfect here

![screenshot from 2018-03-26 04 06 43](https://user-images.githubusercontent.com/7015220/37880912-631cabe4-30ad-11e8-9c63-6adeb6d211fc.png)
<br>
---

* Testimonials slider is not working

![screenshot from 2018-03-26 04 06 15](https://user-images.githubusercontent.com/7015220/37880913-63488db8-30ad-11e8-8c91-93136ab7b372.png)

spyshiv commented:
Travis is also failing on this PR. Please fix that

live-wire commented:
@spyshiv Thanks for the detailed review! I will make the changes that you've pointed out. 
It'd be great if you could address the couple of doubts I had:
- The asset image provided for the VQA challenge is like that. (looks like it is selected). Could you please point me towards the correct asset image for that challenge ? Or should I just use a random image for now ? 
- The grid layout of the page seems a little weird (maybe because of `box-sizing`). Whenever I try to assign 12 columns to a row, it flows into the next row. I had to make quite a lot of workarounds because of this. Could you suggest a solution for this ?
 
Apart from this, I'm on the changes that you've suggested  😄 👍

spyshiv commented:
@live-wire
> * The asset image provided for the VQA challenge is like that. (looks like it is selected). Could you please point me towards the correct asset image for that challenge ? Or should I just use a random image for now ?

Please crop that image. or @shivaniprakash95  Please update that image (final image will have no shadow). 

> * The grid layout of the page seems a little weird (maybe because of box-sizing). Whenever I try to assign 12 columns to a row, it flows into the next row. I had to make quite a lot of workarounds because of this. Could you suggest a solution for this ?

It's all about css positioning. Watch this if you can, your doubts will be cleared
https://www.youtube.com/watch?v=kejG8G0dr5U&list=PLqGj3iMvMa4L731ispRfGAabXeRpM4RL6

___

Feel free to ask in case of anything.

live-wire committed:
Updated Header section with suggested changes

live-wire committed:
Updated Header section with suggested changes

live-wire committed:
Updated partners section

live-wire committed:
Updated partners section

live-wire committed:
Updated the rules section

live-wire committed:
Updated the rules section

live-wire committed:
Updated the Challenges section

live-wire committed:
Updated the Challenges section

live-wire committed:
Footer and testimonial updated

live-wire committed:
Footer and testimonial updated

live-wire commented:
@spyshiv @deshraj 
I made the following(checked) changes in this Pull Request:

- [x] Moved all inline styles to the main style file
- [x] Removed the horizontal scroll from the page
- [x] The page is now better responsive than before
- [x] Made the featured challenges hoverable
- [x] Implemented the carousel animation for Testimonials
- [x] Basic e2e test cases ~**_TODO_**~
- [x] Travis Fix  ~**_TODO_**~

Could you review it while I work on the pending stuff ? 🎈

deshraj commented:
@live-wire: Thanks for the changes. While trying to review, I got the following error:

```
deshraj at macintosh in ~/Projects/EvalAI-ngx on review-47
$ ng serve
Your global Angular CLI version (1.6.6) is greater than your local
version (1.6.5). The local Angular CLI version is used.

To disable this warning use "ng set --global warnings.versionMismatch=false".
** NG Live Development Server is listening on localhost:4200, open your browser on http://localhost:4200/ **
Date: 2018-04-03T05:58:54.935Z
Hash: 4054226a7b13dd09aafa
Time: 1928ms
chunk {inline} inline.bundle.js (inline) 5.79 kB [entry] [rendered]
chunk {main} main.bundle.js (main) 2.93 kB [initial] [rendered]
chunk {polyfills} polyfills.bundle.js (polyfills) 636 bytes [initial] [rendered]
chunk {styles} styles.bundle.js (styles) 87.1 kB [initial] [rendered]
chunk {vendor} vendor.bundle.js (vendor) 853 kB [initial] [rendered]

ERROR in src/app/home/home.component.ts(63,3): error TS1068: Unexpected token. A constructor, method, accessor, or property was expected.
src/app/home/home.component.ts(113,1): error TS1128: Declaration or statement expected.


webpack: Failed to compile.
```

Can you please fix this error and let me know? I can review it again soon.

live-wire committed:
Fixed the ng serve error

live-wire committed:
Fixed the ng serve error

live-wire commented:
@deshraj I fixed the ng serve error. Please review when you find time 👍

live-wire committed:
Fixed Travis errors to get a green light

live-wire committed:
Fixed Travis errors to get a green light

live-wire committed:
Added a couple of e2e tests

live-wire committed:
Added a couple of e2e tests

gautamjajoo commented:
@live-wire 1. As i can see there is a margin on the sides of hte webpage, I think so that should be removed.
2.  The landing page would be till the end of the video.
3. The logos at the bottom of are not properly aligned.
4. The font of the whole webpage is quite big, I think it should be smaller than currently they are.
@deshraj @spyshiv Are these okay? Or they need some changes?

live-wire commented:
Hi @gautamjajoo, Thanks for the feedback
> 1. As i can see there is a margin on the sides of hte webpage, I think so that should be removed.

There is a `marging:8px` on the body tag. I didn't want to touch the global css for just the home page. But yeah, I agree with you on this.
> 2. The landing page would be till the end of the video.
> 4. The font of the whole webpage is quite big, I think it should be smaller than currently they are.

I used the font-size mentioned in this [spec](https://github.com/live-wire/EvalAI-ngx/blob/new-home-page/src/assets/images/EvalAI_Redesign_documentation.jpg). I'm not really convinced about this one :|
> 3. The logos at the bottom of are not properly aligned.

True that! On it!

live-wire committed:
Fixed Alignment of icons in the footer

live-wire committed:
Fixed Alignment of icons in the footer

deshraj commented:
@live-wire I still seem to be facing the webpack error. That's odd I know. Can you tell me what version of node and npm you are using?

live-wire commented:
@deshraj Damn! Can you please share your error log 😞 ?
I'm using node `v8.4.0` and npm `5.7.1`
I'll update my node installation as well.

deshraj commented:
Sorry, it looked like the error was due to the previous npm_modules. I removed it and tried installing packages again and it's working now.

spyshiv commented:
@live-wire 
Please merge latest code from `master` branch before proceeding further.
UI Review:
Please refer screenshot and follow the points given after.
![screencapture-localhost-4200-2018-04-12-16_21_32](https://user-images.githubusercontent.com/7015220/38647027-e289e5ec-3e07-11e8-9966-856e6e190c8e.png)

1. On hovering buttons are lifted upwards that's not good as for as UX point of view(try hovering from buttom of button). Generally if there is a clickable element (button, hyperlink etc.) then on hover it should not move anywhere however you can do operations on hover such as- scale to larger size, highlight border, fill/empty background color, change text color etc. and also animation on all the buttons should be consistent.
2. Replace this Image with a sample youtube video(use `iframe`).  The existing gradient background behind the video is mandatory so do not remove that.
3. The aspect ratio of orgs. logos is not consistent. use `height: auto` that will fix the aspect ratio.
4. Divide this section into `50% +50%`, (currently it is `45% + 50%`)
5. Animation on slide change should be consistent. On changing slide some elements are going to left/right and some are fading out.
6. Set height of this input equal to height of `get notification` button.

spyshiv commented:
@live-wire 
Addional review:
* You should look into `variable.scss` and `base.scss` to use relevant class into your html code according to font sizes. i.e. use css class `fs-14` for `font-size: 14px`.
* Use theme colors instead of choosing colors by own, refer to `variables.scss`.
* In html files, you used class `row` and didn't use column class in children elements of it. Please use grid system properly. refer to `grid.scss`

References:
`grid.scss` > `/src/styles/grid.scss`
`variables.scss` > `/src/styles/variables.scss`

PS: In the design docs the font size parameters are based on photoshop font renderer that's why fonts are larger than recommended size. Standard range is `12px-36px` that is defined in `variables.scss` please use that instead of given range(`20px-60px` acc. to issue #46). 
@aka-jain @shivaniprakash95 Need your suggestions here

please let me know in case of anything.

live-wire commented:
@spyshiv Thank you for the detailed feedback and steps 👍 I'll try to wrap it up as soon as possible.

deshraj commented:
Thanks @live-wire for taking care of this. We appreciate your effort. :)

live-wire committed:
Merge branch 'new-home-page' of https://github.com/live-wire/EvalAI-ngx into new-home-page

live-wire committed:
Header and Partner sections updated

live-wire committed:
Updated Rules Testimonial and Footer

live-wire committed:
Minor CSS update

live-wire commented:
@spyshiv I made the following changes after your suggestions:
- Button animations and heights fixed
- I found [this](https://www.youtube.com/watch?v=rWiycwJIXh4&t=37s) video by @gautamjajoo for the iframe 😄
- Used `height:auto` for org-logos like you suggested ⚡️ 
- Fixed the rules section and used the _grid_ there
- Used the `fade` animation for all elements in testimonials
- Used theme colors and font-sizes from `variables.scss`
- Updated the font of the page to the following mapping:
**Spec -> Actual** 
```
20px -> 12px
30px -> 18px
40px -> 24px
60px -> 36px
``` 

Please have a look and let me know about further improvements 😃 🍕 🍺

gautamjajoo commented:
@live-wire The github logo and the location pin should be changed. 
Originally it is


![github-logo](https://user-images.githubusercontent.com/24366008/38724885-23ddfd32-3f23-11e8-9f32-ecc014b610ca.png)

live-wire committed:
Updated git icon and location marker

live-wire commented:
@spyshiv @deshraj Travis failed with the following error:
`No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.` 
Apparently _surge_ expects inputs(like email) from the user that it didn't recieve and the travis build failed due to timeout.

deshraj committed:
Merge branch 'master' into new-home-page

aka-jain commented:
We should start developing every component in different PR's. So closing this for now. 
cc @live-wire @deshraj
